### ISSUE TITLE
[Bug]: predefinedDataTemplates not working

### ISSUE THREAD

**Issue by passioneight** [pimcore/pimcore#12737]
### Expected behavior

Using a `pimcore_image` with the following configuration allows to add predefined data templates for hotspots:

```twig
{{ pimcore_image("image", {
    'thumbnail': 'my-thumbnail',
    'class': 'my-class',
    'predefinedDataTemplates': {
        'hotspot': [
            {
                'menuName': 'hotspot config 1',
                'name': 'hotspot name',
                'data': [
                    {
                        'name': 'my textfield',
                        'type': 'textfield'
                    }
                ]
            }
        ]
    }
}) }}
```

### Actual behavior

Nothing happens.

### Steps to reproduce

1. Add the editable
2. Try to find a UI-element that allows you to add the `hotspot config 1`

**Comment by passioneight** [comment]
Originally commented in https://github.com/pimcore/pimcore/pull/2554#issuecomment-1192392504 .

**Comment by passioneight** [comment]
Here's a patch, will create a PR when I've some time:

```patch
Index: public/bundles/pimcoreadmin/js/pimcore/element/tag/imagehotspotmarkereditor.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/public/bundles/pimcoreadmin/js/pimcore/element/tag/imagehotspotmarkereditor.js b/public/bundles/pimcoreadmin/js/pimcore/element/tag/imagehotspotmarkereditor.js
--- a/public/bundles/pimcoreadmin/js/pimcore/element/tag/imagehotspotmarkereditor.js	
+++ b/public/bundles/pimcoreadmin/js/pimcore/element/tag/imagehotspotmarkereditor.js	(date 1658486250531)
@@ -679,29 +679,27 @@
                 callbackFunction();
             }.bind(this)
         };
-
-        if (this.predefinedDataTemplates[type] && this.predefinedDataTemplates[type].length > 0) {
+        
+        if (this.predefinedDataTemplates[type] && Object.keys(this.predefinedDataTemplates[type]).length > 0) {
             buttonConfig.xtype = "splitbutton";
             var menu = [];
-            for (var i = 0; i < this.predefinedDataTemplates[type].length; i++) {
-                var templateConfig = this.predefinedDataTemplates[type][i];
+
+            Object.values(this.predefinedDataTemplates[type]).forEach(templateConfig => {
                 var templateConfigName = templateConfig.name;
-                var templateMenuName = templateConfig.menuName ? templateConfig.menuName : templateConfigName;
-                if (!templateConfigName) {
-                    templateConfigName = "&nbsp";
-                }
+                var templateMenuName = templateConfig.menuName ?? templateConfigName;
+
                 menu.push(
                     {
                         text: t(templateMenuName),
                         iconCls: "pimcore_icon_hotspotmarker_template",
                         handler: function (templateConfig) {
                             var elId = callbackFunction(templateConfig);
-                            var copiedData = templateConfig.data ? templateConfig.data.slice() : [];
-                            this.hotspotMetaData[elId] = copiedData;
+                            this.hotspotMetaData[elId] = templateConfig.data ? templateConfig.data.slice() : [];
                         }.bind(this, templateConfig)
                     }
                 );
-            }
+            });
+
             buttonConfig.menu = menu;
         }
```

**Comment by dvesh3** [comment]
Fixed by #12741
